typescript/db/src/repository/image-processing.ts

```diff
@@ -103,21 +104,40 @@ export const processImage: Repository["imageProcessing"]["processImage"] =
     const buffer = await getImage(url);
     const image = await Jimp.read(buffer as Buffer);
 
-    const generateThumbnailFromSize = (size: ThumbnailSizes) =>
-      generateThumbnail({
-        size,
-        image,
-        url,
-        putImage,
-      });
+    const { url: thumbnail500UrlNew, image: imageSrc } = thumbnail500Url
+      ? { url: thumbnail500Url, image }
+      : ((await generateThumbnail({ size: 500, image, url, putImage })) as {
+          url: string;
+          image: Jimp;
+        });
```

Comment:
The `as` was hiding a bug...
`image` can be `undefined` as the `catch` clause of `generateThumbnail` returns `undefined`. I've fixed it during the cleanup I've pushed over your commit 😉